qlib/data/ops.py

```diff
@@ -96,6 +98,37 @@ def _load_internal(self, instrument, start_index, end_index, *args):
         return self.feature.load(instrument, start_index, end_index, *args)
 
 
+class Cat(ElemOperator):
```

Comment:
> Did you encounter any performance issues if we force convert it to category every time?

No performance issue. The reason for adding the `Cat` operator is that every time you take any field, you need to determine whether the field is of category type, which affects non-category type fields. 

It is a good suggestion to handle category data logic in `FileFeatureStorage`, which will make filestorage clearer.

Adding the function to get the field type in `FeatureProvider` does not seem to have practical use at present, and it will be added when it is useful.